include/modem/gcf_sms_lib.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef GCF_SMS_LIB_H_
+#define GCF_SMS_LIB_H_
```

Comment:
Let's get rid of this file.

Also, I would call this library `gfc_sms` with the `lib` part at the end, wherever it is applicable, it is redundant (`gcf_sms_lib` library vs `gcf_sms` library).